ambari/datadog_checks/ambari/ambari.py

```diff
@@ -0,0 +1,198 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from six import iteritems
+from requests.auth import HTTPBasicAuth
+import requests
+
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.errors import CheckException
+from . import common
+
+# Tag templates
+CLUSTER_TAG_TEMPLATE = "ambari_cluster:{}"
+HOST_TAG = "ambari_host:"
+SERVICE_TAG = "ambari_service:"
+COMPONENT_TAG = "ambari_component:"
+
+# URL queries
+COMPONENT_METRICS_QUERY = "/components?fields=metrics"
+SERVICE_INFO_QUERY = "?fields=ServiceInfo"
+
+# Response fields
+METRICS_FIELD = "metrics"
+
+
+class AmbariCheck(AgentCheck):
+    def __init__(self, *args, **kwargs):
+        super(AmbariCheck, self).__init__(args, kwargs)
+        self.hosts = []
+        self.clusters = []
+        self.services = []
+
+    def check(self, instance):
+        server = instance.get("url", "")
+        port = str(instance.get("port", ""))
+        path = str(instance.get("path", ""))
+        base_tags = instance.get("tags", [])
+        whitelisted_services = instance.get("services", [])
+        authentication = {'username': instance.get("username", ""),
```

Comment:
I believe you can remove authentication from everywhere, the http wrapper will do the magic.